actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -2,8 +2,14 @@ module ActionController
   module ImplicitRender
     def send_action(method, *args)
       ret = super
-      default_render unless performed?
-      ret
+      return ret if performed?
+
+      if template_exists?(method, _prefixes)
```

Comment:
I propose as leaving `method_for_action` alone. The current method body is:

```ruby
    def method_for_action(action_name)
      super || if template_exists?(action_name.to_s, _prefixes)
        "default_render"
      end
    end
```

The reason I say this is that the default behavior of `default_head` should apply when there is a valid method but not in the case of there being no method at all. This could lead to more complex errors such as getting `head :no_content` that doesn't seem to apply to a method.